### PR TITLE
Refactor meshok-react-hooks TodoItem

### PR DESCRIPTION
**Pull request by codewylle**

### CODE REVIEW

File: packages/meshok-react-hooks/src/components/TodoItem.tsx [hunk 1]
```diff
@@ -11,24 +11,33 @@ export function TodoItem({ itemData }: TodoItemProps): JSX.Element {
...
+	}
+
+	function renderChildren(children: ITodo[] | undefined): JSX.Element {
+		return (children && children.length > 0)
+			?	<ul>{
+					children.map((child, i) =>
+						<li key={i}><TodoItem itemData={child} /></li>
+					)
+				}</ul>
+			: <></>
```
Comment (finico) [inline packages/meshok-react-hooks/src/components/TodoItem.tsx]: `null`

File: packages/meshok-react-hooks/src/components/TodoItem.tsx [hunk 2]
```diff
@@ -11,24 +11,33 @@ export function TodoItem({ itemData }: TodoItemProps): JSX.Element {
...
+		title: string | undefined,
+		isEditing: boolean,
+	): JSX.Element {
+		return title
+			? isEditing
+				?	<label>
+						<input type="text" defaultValue={itemData.title} />
+					</label>
+				: <label>{itemData.title}</label>
+			: <></>
```
Comment (finico) [inline packages/meshok-react-hooks/src/components/TodoItem.tsx]: `null`

File: packages/meshok-react-hooks/src/components/TodoItem.tsx [hunk 3]
```diff
@@ -11,24 +11,33 @@ export function TodoItem({ itemData }: TodoItemProps): JSX.Element {
...
+	}
+
+	function renderChildren(children: ITodo[] | undefined): JSX.Element {
```
Comment (finico) [inline packages/meshok-react-hooks/src/components/TodoItem.tsx]: Why not just `(children?: ITodo[])` ?

File: packages/meshok-react-hooks/src/components/TodoItem.tsx [hunk 4]
```diff
@@ -11,24 +11,33 @@ export function TodoItem({ itemData }: TodoItemProps): JSX.Element {
...
-						</li>
-					))}
-				</ul>
-			)}
+			{renderTitle(itemData.title, isEditing)}
+			{renderChildren(itemData.children)}
 		</>
 	);
+
+	// WHERE
```
Comment (finico) [inline packages/meshok-react-hooks/src/components/TodoItem.tsx]: What is that means "where"?

File: packages/meshok-react-hooks/src/components/TodoItem.tsx (hunk 2 above)
Comment (codewylle) [reply packages/meshok-react-hooks/src/components/TodoItem.tsx]: I am meshok. JSX.Element may be null

Comment (codewylle) [reply packages/meshok-react-hooks/src/components/TodoItem.tsx]: Or Not =)

**Comment by codewylle** [comment]
I have listened to you

**Comment by codewylle** [comment]
Merge to be8b3ce30c41d1ebf09d782639cb70cff68ce01a

**Pull request merged**